core/manager/runtime_manager.go

```diff
@@ -49,27 +50,15 @@ func NewKogitoRuntimeManager(context operator.Context, runtimeHandler KogitoRunt
 // FetchKogitoRuntimeDeployments gets all dcs owned by KogitoRuntime services within the given namespace
 func (k *kogitoRuntimeManager) FetchKogitoRuntimeDeployments(namespace string) ([]v1.Deployment, error) {
 	var kdcs []v1.Deployment
-	kogitoRuntimeServices, err := k.runtimeHandler.FetchAllKogitoRuntimeInstances(namespace)
-	if err != nil {
-		return nil, err
-	} else if len(kogitoRuntimeServices.GetItems()) == 0 {
-		return kdcs, nil
-	}
-
 	deploymentHandler := infrastructure.NewDeploymentHandler(k.Context)
 	deps, err := deploymentHandler.FetchDeploymentList(namespace)
 	if err != nil {
 		return nil, err
 	}
 	k.Log.Debug("Looking for Deployments owned by KogitoRuntime")
 	for _, dep := range deps.Items {
-		for _, owner := range dep.OwnerReferences {
-			for _, app := range kogitoRuntimeServices.GetItems() {
-				if owner.UID == app.GetUID() {
-					kdcs = append(kdcs, dep)
-					break
-				}
-			}
+		if util.MapContains(dep.Annotations, operator.KogitoRuntimeKey, "true") {
```

Comment:
If there will be multiple kogito Deployments then all will be returned, even though they don't belong to specific KogitoRuntime. Is it expected?

Reply:
I see it is expected, can be closed.